js/src/dashboard-widget.js

```diff
@@ -51,8 +52,10 @@ class DashboardWidget extends Component {
 		this.getStatistics();
 		this.getFeed();
 
-		this.setState( {
-			isDataFetched: true,
+		domReady( () => {
+			this.setState( {
+				isDataFetched: true,
+			} );
 		} );
```

Comment:
You can implement the special `componentDidMount` [lifecycle method](https://reactjs.org/docs/state-and-lifecycle.html#adding-lifecycle-methods-to-a-class) to run code after the React component has been mounted.

I would move the code that fetches the data from the constructor to a new `componentDidMount` method like this:
```js
componentDidMount() {
	const widgetCheckbox = jQuery( "#wpseo-dashboard-overview-hide" );

	// Only fetch the data if the Yoast SEO dashboard widget is checked in the Screen Options.
	if ( widgetCheckbox.is( ":checked" ) ) {
		this.fetchData();
	}

	// Whenever the checkbox gets clicked, fetch the data if needed.
	widgetCheckbox.on( "click", () => {
		this.fetchData();
	} );
}
```

Reply:
The `domReady` would not be necessary anymore 🙂.